### PR TITLE
UAVCAN battery support

### PR DESCRIPTION
**Pull request by dakejahl**
~This is not working yet, opening to get some feedback/review.~

### CODE REVIEW

**Comment by dagar** [comment]
Looks good. Starting with a single battery should be fine, but this will likely need to extend to multiple.

The CDev stuff within UAVCAN sensors needs to be killed off entirely, but it looks like it's mostly out of your way for batteries.

**Comment by dakejahl** [comment]
This works. Yes the plan is to extend this as soon as it becomes necessary.

File: src/drivers/uavcan/sensors/battery.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2014, 2015 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/uavcan/sensors/battery.hpp]: year

File: src/drivers/uavcan/sensors/battery.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	int res = device::CDev::init();
+
+	if (res < 0) {
+		return res;
+	}
+
+	res = _sub_battery.start(BatteryInfoCbBinder(this, &UavcanBatteryBridge::battery_sub_cb));
+
+	if (res < 0) {
+		DEVICE_LOG("failed to start uavcan sub: %d", res);
```
Comment (dagar) [inline src/drivers/uavcan/sensors/battery.cpp]: PX4_ERR here might actually be useful.

**Pull request merged**